src/OpenTelemetry/Metrics/AggregatorStore.cs

```diff
@@ -51,11 +51,13 @@ internal sealed class AggregatorStore
         internal AggregatorStore(
             AggregationType aggType,
             AggregationTemporality temporality,
+            int metricPointLimit,
             double[] histogramBounds,
             string[] tagKeysInteresting = null)
         {
-            this.metricPoints = new MetricPoint[MaxMetricPoints];
-            this.currentMetricPointBatch = new int[MaxMetricPoints];
+            this.metricPointLimit = metricPointLimit;
+            this.metricPoints = new MetricPoint[metricPointLimit];
```

Comment:
I think metricPoinLimit+1 should be used, as we reserve one point for zero dimension case.

Reply:
Would it not be misleading for the user if we add +1 here? Maybe we could update the remarks for the public method `SetMetricPointPerMetricStreamLimit` in that case?

Reply:
> I think metricPoinLimit+1 should be used, as we reserve one point for zero dimension case.

That doesn't seem to be mathematically correct?

Reply:
We would throw if the provided value is less than 1 as the method validates the input:
`Guard.Range(metricPointLimit, min: 1);`

I think what could be confusing is that the user provides a value thinking that it will be maximum number of time-series allowed but internally we would be allowing an additional time-series.

Reply:
Should we consider doing away with the reservation of the zero dimension case? I guess I don't know the history of why we do it.